Assignment6.Rmd

```diff
@@ -52,32 +50,47 @@ post(c.tree1, file = "tree1.ps", title = "MOOC") #This creates a pdf image of th
 #If we are worried about overfitting we can remove nodes form our tree using the prune() command, setting cp to the CP value from the table that corresponds to the number of nodes we want the tree to terminate at. Let's set it to two nodes.
 
 ```{r}
-c.tree2 <- prune(c.tree1, cp = )#Set cp to the level at which you want the tree to end
+c.tree2 <- prune(c.tree1, cp = 0.058182)#Set cp to the level at which you want the tree to end
 
 #Visualize this tree and compare it to the one you generated earlier
-
+rpart.plot(c.tree2)
 post(c.tree2, file = "tree2.ps", title = "MOOC") #This creates a pdf image of the tree
 ```
 
 #Now use both the original tree and the pruned tree to make predictions about the the students in the second data set. Which tree has a lower error rate?
 
 ```{r}
 M2$predict1 <- predict(c.tree1, M2, type = "class")
-
 M2$predict2 <- predict(c.tree2, M2, type = "class")
-
-table(M2$certified, M2$predict1)
-
-table(M2$certified, M2$predict2)
-
+TB1<-table(M2$certified, M2$predict1)
+TB2<-table(M2$certified, M2$predict2)
+P1<-(TB1[2,2]+TB1[1,1])/sum(TB1)
+P2<-(TB2[2,2]+TB2[1,1])/sum(TB2)
+P1
+P2
+#c.tree2 (prediction2) has a lower error rate
 ```
 
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
-
+library(readr)
+D <- read_csv("student.course.csv")
+#Model1:
+c.tree3<- rpart(GRD_PTS_PER_UNIT~ GPAO+TERM, data=D)
+printcp(c.tree3)
+library(dplyr)
+#Model2:
+D1<-select(D, -SUBJECT, -DIV)
+D1<-scale(D1, center=TRUE)
+pca<-prcomp(D1,scale=TRUE)
+D2<-data.frame(pca$x)
+D3<- bind_cols(D, D2)
+c.tree4<-rpart (GRD_PTS_PER_UNIT~ PC3+PC6, data=D3)
+printcp(c.tree4)
+#choose model1 because it has a lower error
```

Comment:
Generally you did a good work. It seems you understand the basics of model comparison. When making the decision, it is better to report the specific measures/numbers rather than just saying which one is lower. In addition, cv error rate is one important factor to consider when selecting models, think of other factors that could also be considered.